### ISSUE TITLE
the words end with number always throw error

### ISSUE THREAD

**Issue by misogihagi** [kn1cht/textlint-rule-use-si-units#20]
#19

**Comment by kn1cht** [comment]
@misogihagi Thank you for reporting this issue!
I ran your tests and took a measure to prevent false-positives.

## 我ら5人合わせて四天王！
This test passes because this rule does not detect Japanese characters as possible units.
For example, this rule does not raise any error to '100垓' or '秒速5㌢㍍'.

## The 5 of us together are the Four Heavenly Kings!
This test has failed with a message: `「of」には、SI単位系で使用できない文字が含まれています。SI単位系を使用してください。`
This is a false positive and "of" should not be warned.

When I wrote this rule, my idea was to detect all possible "number + unit" and exclude words which are not units using a white list.

https://github.com/kn1cht/textlint-rule-use-si-units/blob/290f4d48d8d061eaa4b4dee97b5b7574949d7640/src/use-si-units.js#L16

Since it turned out that this list was inadequate, I added more words which may be placed after numbers (thanks to ChatGPT).

https://github.com/kn1cht/textlint-rule-use-si-units/blob/68abb9d73c529eeb32ce88a937416baafe8e1fd1/src/use-si-units.js#L16

It is open to debate whether the whitelist approach is a good idea or not.
It is difficult to prevent false positives completely (e.g., "2 in" should be warned as an expression using an inch that is not an SI unit, but "2 in 1" is not).
If you have any good ideas, further pull requests are welcome.

**Comment by kn1cht** [comment]
If you have to prevent such unwanted errors before this rule is improved, you can use `allowedUnits` option.

```
{
    "rules": {
        "use-si-units": {
            "allowedUnits": ["of", "in", ...],
        }
    }
}
```

**Comment by kn1cht** [comment]
I published v2.0.0!

https://www.npmjs.com/package/textlint-rule-use-si-units
